apps/api/src/app/auth/usecases/login/login.usecase.ts

```diff
@@ -41,7 +41,8 @@ export class Login {
       throw new UnauthorizedException(`Account blocked, Please try again after ${blockedMinutesLeft} minutes`);
     }
 
-    if (!user.password) throw new ApiException('OAuth user login attempt');
+    // TODO: Trigger a password reset flow automatically for existing OAuth users instead of throwing an error
+    if (!user.password) throw new ApiException('Please sign in using Github.');
```

Comment:
Note that returning different errors per auth flow results in User enumeration issues. User enumeration is mostly a privacy concern and should be tackled in a separate PR.

Reply:
Moreover, the error message mentions Github specifically. Currently, it's the only supported OAuth method. When more methods are added, the copywriting should be tweaked.

Reply:
Btw we have the Google OAuth logic but currently, it's enterprise-only and "disabled".